Packages/Core/Editor/Generator/Generator.cs

```diff
@@ -1,164 +1,232 @@
+        /// Generate scripts inside the generators directory from the generators templates.
         /// </summary>
-        /// <param name="template">The content template to remove namespace from.</param>
-        /// <returns>A copy of `content`, but without duplicate namespaces.</returns>
-        private static string RemoveDuplicateNamespaces(string template)
+        public void Generate()
         {
-            var currentIndex = 0;
-            var countNamespaces = new Dictionary<string, int>();
-            while (true)
+            var assetPaths = new string[solvers.Values.Count];
+
+            try
             {
-                currentIndex = template.IndexOf("using ", currentIndex);
-                if (currentIndex == -1) break;
-                var namespaceStartIndex = currentIndex + 6;
-                var namespaceEndIndex = template.IndexOf(";", namespaceStartIndex);
-                var ns = template.Substring(namespaceStartIndex, namespaceEndIndex - namespaceStartIndex);
-                if (countNamespaces.ContainsKey(ns))
+                AssetDatabase.StartAssetEditing();
+
+                for (int i = 0; i < solvers.Values.Count; i++)
                 {
-                    countNamespaces[ns] = countNamespaces[ns] + 1;
+                    var solver = solvers.Values[i];
+
+                    if (solver.option)
+                    {
+                        var directory = solver.script
+                            ? AssetDatabase.GetAssetPath(solver.script)
+                            : AssetDatabase.GetAssetPath(this);
+                        directory = Path.GetDirectoryName(directory);
+
+                        Generate(directory, solver.template);
+                        assetPaths[i] = $"{directory}/{solver.template.className}.cs";
+                    }
                 }
-                else
+            }
+            finally
+            {
+                AssetDatabase.StopAssetEditing();
+                AssetDatabase.SaveAssets();
+            }
+
+            for (int i = 0; i < solvers.Values.Count; i++)
+            {
+                var solver = solvers.Values[i];
+
+                if (solver.option)
                 {
-                    countNamespaces.Add(ns, 1);
+                    solvers.Values[i].script = AssetDatabase.LoadAssetAtPath<MonoScript>(assetPaths[i]);
                 }
-                currentIndex = namespaceEndIndex;
             }
+        }
+
+        /// <summary>
+        /// Generate scripts inside the directory from the input templates.
+        /// </summary>
+        /// <param name="directory">The folder path where the scripts are going to be written to.</param>
+        /// <param name="template">The template to generate the script from.</param>
+        /// <param name="templates">The templates to generate the scripts from.</param>
+        public static MonoScript[] Generate(string directory, params Template[] templates)
```

Comment:
I would like to keep it in there as a general method. The idea behind it is that if you have a Generate(string, Template) method somewhere and ever want to extend it, you can just do Generate(string, Template, Template, Template) and it will work.

Reply:
Alright, that is fine